### DISCUSSION TITLE
Mirth restart actions

### DISCUSSION THREAD

**Discussion by swapnilBR** [nextgenhealthcare/connect#4662] [Q&A]
Hello, I want to know the actions performed internally on Mirth(mcservice.exe) restart. Does it include something like cleaning some cache's, shutting down SQL connection's, etc.? 
Any leads on this would be helpful since I'm looking into a issue with custom application being deployed in Mirth and the issue gets resolved on Mirth restart. So, I need to understand the Mirth Restart process.

Thanks in advance! :)

**Comment by jonbartels** [comment]
https://github.com/nextgenhealthcare/connect/blob/924d1860bcb5f01fe125edcfae183d7410e61976/server/src/com/mirth/connect/server/Mirth.java#L404 is a good place to start. The `mcservice` script eventually calls this function. 

Start there and come back with any specific questions.

If you are experiencing a specific problem with your custom application it may also be productive to describe that problem and any errors you are encountering.

**Reply by jonbartels** [reply]
The question was crossposted to https://forums.mirthproject.io/forum/mirth-connect/support/179510-mirth-restart-actions
